olp-cpp-sdk-dataservice-read/tests/QueryApiTest.cpp

```diff
@@ -0,0 +1,256 @@
+constexpr auto kQuadKey = "23618401";
+constexpr auto kNodeBaseUrl =
+    "https://some.node.base.url/stream/v2/catalogs/"
+    "hrn:here:data::olp-here-test:hereos-internal-test-v2";
+constexpr auto kUrlQuadTreeIndexVolatile =
+    R"(https://some.node.base.url/stream/v2/catalogs/hrn:here:data::olp-here-test:hereos-internal-test-v2/layers/test-layer/quadkeys/23618401/depths/2)";
+
+constexpr auto kUrlQuadTreeIndexVolatileAllInputs =
+    R"(https://some.node.base.url/stream/v2/catalogs/hrn:here:data::olp-here-test:hereos-internal-test-v2/layers/test-layer/quadkeys/23618401/depths/2?additionalFields=checksum%2CdataSize&billingTag=OlpCppSdkTest)";
+
+constexpr auto kHttpResponseEmpty = R"jsonString()jsonString";
+constexpr auto kHttpResponseQuadTreeIndexVolatile =
+    R"jsonString({ "parentQuads": [ { "additionalMetadata": "string", "checksum": "string", "compressedDataSize": 0, "dataHandle": "675911FF6236B7C7604BF8B105F1BB58", "dataSize": 0, "crc": "c3f276d7", "partition": "73982", "version": 0 } ], "subQuads": [ { "additionalMetadata": "string", "checksum": "291f66029c232400e3403cd6e9cfd36e", "compressedDataSize": 200, "dataHandle": "1b2ca68f-d4a0-4379-8120-cd025640510c", "dataSize": 1024, "crc": "c3f276d7", "subQuadKey": "string", "version": 1 } ] })jsonString";
+
+using testing::_;
+using namespace ::testing;
+using namespace olp;
+using namespace olp::client;
+using namespace olp::dataservice::read;
+using namespace olp::tests::common;
+
+class QueryApiTest : public Test {
+ protected:
+  void SetUp() override {
+    network_mock_ = std::make_shared<NetworkMock>();
+
+    settings_ = std::make_shared<OlpClientSettings>();
+    settings_->network_request_handler = network_mock_;
+
+    client_ = OlpClientFactory::Create(*settings_);
+    client_->SetBaseUrl(kNodeBaseUrl);
+  }
+  void TearDown() override { network_mock_.reset(); }
+
+ protected:
+  std::shared_ptr<OlpClientSettings> settings_;
+  std::shared_ptr<OlpClient> client_;
+  std::shared_ptr<NetworkMock> network_mock_;
+};
+
+TEST_F(QueryApiTest, QuadTreeIndexVolatile) {
+  {
+    SCOPED_TRACE("Request metadata for tile.");
+    EXPECT_CALL(
+        *network_mock_,
+        Send(AllOf(IsGetRequest(kUrlQuadTreeIndexVolatile)), _, _, _, _))
+        .WillOnce(ReturnHttpResponse(
+            http::NetworkResponse().WithStatus(http::HttpStatusCode::OK),
+            kHttpResponseQuadTreeIndexVolatile));
+
+    int32_t depth = 2;
+    auto index_response =
+        olp::dataservice::read::QueryApi::QuadTreeIndexVolatile(
+            *client_, kLayerid, kQuadKey, depth, boost::none, boost::none,
+            olp::client::CancellationContext{});
+
+    ASSERT_TRUE(index_response.IsSuccessful());
+    auto result = index_response.GetResult();
+    ASSERT_EQ(1u, result.GetSubQuads().size());
+    ASSERT_EQ(1u, result.GetParentQuads().size());
+  }
+  {
+    SCOPED_TRACE(
+        "Request metadata with billing tag and aditional additional fields.");
```

Comment:
```suggestion
        "Request with billing tag + additional fields.");
```

This log is for developers, you do not need to make it perfect grammar. When this test fails you only need to know which of the cases fail and what the case does.